components/apps.js

```diff
@@ -516,11 +530,12 @@ SteamUser.prototype._getLicenseInfo = async function() {
 		return;
 	}
 
-	let packageids = this.getOwnedPackages();
+	// Get all owned lisense id's
+	let packageids = this.licenses.map(license => license.package_id);
```

Comment:
This PR pretty much splits the definition of owned licenses and owned packages. Licenses are all the packages tied to the account. Owned packages are licenses without expired (and shared & temporary) licenses. 
Additionally, retrieving owned packages can now only be done after `appOwnershipCached` is emitted. Retrieving owned licenses can trivially still be done after `licenses` is emitted.

Reply:
Perhaps you could add a `getOwnedLicenses` function which can be called as soon as `licenses` event has been emitted, which pretty much just returns `this.licenses`, but I leave that to you. Not sure if you want that.

Reply:
I'm kind of thinking `getOwnedPackages` should maintain its present behavior, but add an `options` object argument to exclude things. Like:

```js
this.getOwnedPackages({
    excludeExpired: true,
    excludeTemporary: true,
    excludeShared: true
});
```

Reply:
It already has an `options` object, but with different property names. Personally, I omitted using include/exclude in the keynames, to make it less redundant, but I am more than happy to change the keynames to your suggestion. Maintaining the present behavior is easier said than done, but I have an idea. I could add `getOwnedLicenses()` (as suggested before) and return it when `getOwnedPackages` is called without an `options` argument (like before).

Reply:
I'm not married to those key names, in fact I kind of prefer the way you did it. I wrote that comment before reading that part of the code.

Reply:
Check new commit

Reply:
> I'm not married to those key names, in fact I kind of prefer the way you did it. I wrote that comment before reading that part of the code.

Well, I read this too late. Oops. On the bright side, it did help to remove many `if (!boolean)` statements.